en/docs/api-analytics/samples/publishing-custom-analytics-data.md

```diff
@@ -0,0 +1,193 @@
+        </exclusions>
+    </dependency>
+
+!!! Info
+
+	- The versions for ${carbon.apimgt.version}, and ${synapse.version} can be found in the jar versions available in the current API Manager 4.0.0 package.
+    - Since this feature support available in the latest update levels, you have to follow the below steps to add the required artifacts to the local m2 repository and point it as a repository in pom file.
+        1. Find the `org.wso2.carbon.apimgt.gateway` and `org.wso2.carbon.apimgt.common.analytics` jars inside the `<APIM_HOME>/repository/components/plugins` directory in latest U2 updated APIM pack.
+        2. Add the above jars to the local m2 manually.
+        ```code
+        mvn install:install-file -Dfile=<PATH_TO_FILE>/org.wso2.carbon.apimgt.gateway_<COMPENENT_VERSION>.jar -DgroupId=org.wso2.carbon.apimgt -DartifactId=org.wso2.carbon.apimgt.gateway -Dversion=<COMPENENT_VERSION> -Dpackaging=jar
+        mvn install:install-file -Dfile=<PATH_TO_FILE>/org.wso2.carbon.apimgt.common.analytics_<COMPENENT_VERSION>.jar -DgroupId=org.wso2.carbon.apimgt -DartifactId=org.wso2.carbon.apimgt.common.analytics -Dversion=<COMPENENT_VERSION> -Dpackaging=jar
+        ```
+        3. Point local m2 repository in project pom
+        ```code
+        <repository>
+            <id>local-maven-repo</id>
+            <url>file://home/user/.m2/repository</url>
+        </repository>
+        ```
+        Follow the URL pattern when providing the repo url
+
+### Implementing Required Class
+
+#### AnalyticsCustomDataProvider Implementation Class
+
+Custom analytics data can be added to the existing event schema with the use of a class of type `AnalyticsCustomDataProvider`. Therefore, it is required to implement a class from the `AnalyticsCustomDataProvider` interface. By overriding the `getCustomProperties` method it is possible to push custom analytics data.
+
+In order to achieve this behavior, create a class implementing the `AnalyticsCustomDataProvider` Interface of `org.wso2.carbon.apimgt.common.analytics.collectors`.
+
+Implementation of this class should look something similar to [this](https://github.com/wso2/samples-apim/blob/4.0.0/analytics-custom-data-provider/src/main/java/org/wso2/carbon/apimgt/gateway/sample/publisher/CustomDataProvider.java).
+
+#### Build the Project
+
+Build the project using,
+
+    mvn clean install
+
+## Configuring the Sample
+
+This section will cover the steps required to configure WSO2 API-M Gateway for the sample created above. The steps covered are adding the .jar file, configuring the deployment.toml file, and enabling the logs.
+
+!!! Warning
+    Note that WSO2 API Manager 3.0.0, 3.1.0, 3.2.0, and 4.0.0 are affected by the **Log4j2 zero-day** vulnerability, which has been reported to WSO2 on 10th December 2021. You can mitigate this vulnerability in your product by following our [instructions and guidelines](https://docs.wso2.com/pages/viewpage.action?pageId=180948677).
+
+1. Adding the .jar file.
```

Comment:
Let's say what this .jar file is,

Reply:
Fixed with https://github.com/wso2/docs-apim/pull/6104/commits/1a243d5c12d13171c6a00eecdba75d374c2bc77c
Thanks